modules/preferences/main/index.coffee

```diff
@@ -217,15 +217,18 @@ class Preferences extends hx.EventEmitter
       modal: modal
     }
 
-    defaultLocaleId = moment?.locale() or navigator.languages?[0] or navigator.language
+    defaultLocaleId = navigator.languages?[0] or navigator.language
+
+    @on('localechange', (value) -> moment?.locale(value))
```

Comment:
Rather than adding a listener to itself, this should instead be done when the locale is set in the `locale` method (i.e. when the `localechange` event is emitted)